demo/events.py

```diff
@@ -0,0 +1,19 @@
+from demo.celery import app
+
+
+# noinspection PyUnusedLocal
+@app.event('demo.first.occured')
+def event_occured(value: str):
+    pass
+
+
+@app.event('demo.number.even')
+def number_is_even(number: int):
+    if number % 2 != 0:
+        raise ValueError(number)
+
+
+@app.event('demo.number.odd')
+def number_is_odd(number: int):
+    if number % 2 == 0:
+        raise ValueError(number)
```

Comment:
Where should events be described? 

Only for consumer service or also for producers service?

Reply:
Events signatures are separated from each other on consumer and producer sides, so short answer is "on both sides".
Validation should be done on producer side, and maybe for backward-compatibility purposes consumer should declare something like 
```python
def my_event(*args, **kwargs):
    ...
```

Reply:
:( I hoped you have resolved the problem of multiple declarations of an event.

Reply:
If you intentionally want to unite publisher and consumer event signature declaration, just use common library that stores these signatures. If you don't want to share same dependencies - you are allowed to separate event declaration on server and client sides.